public/js/fxa-analytics.js

```diff
@@ -0,0 +1,101 @@
+/* eslint-env browser */
+"use strict";
+/* global ga */
+/* global ga_getLocation  */
+
+
+
+const hasParent = (el, selector) => {
+  while (el.parentNode) {
+    el = el.parentNode;
+    if (el.id === selector)
+      return el;
+  }
+  return null;
+};
+
+const setMetricsIds = (el) => {
+  if (hasParent(el, "scan-another-email")) {
+    el.dataset.eventCategory = "Scan Another Email Form";
+  }
+  if (el.dataset.fxaEntrypoint && hasParent(el, "sign-up-banner")) {
+    el.dataset.eventCategory = `${el.dataset.eventCategory} - Banner`;
+    el.dataset.fxaEntrypoint = `${el.dataset.fxaEntrypoint}-banner`;
+  }
+  return;
+};
+
+const getLocation = () => {
+  const eventLocation = document.querySelectorAll("[data-page-label]");
+  if (eventLocation.length > 0) {
+    return `Page ID: ${eventLocation[0].dataset.pageLabel}`;
+  } else {
+    return `Page ID: ${ga_getLocation()}`;
+  }
+};
+
+const sendPing = async(el, eventAction, eventLabel = null) => {
+  if (typeof(ga) !== "undefined" && !el.classList.contains("hide")) {
+    if (!eventLabel) {
+      eventLabel = `${getLocation()}`;
+    }
+    const eventCategory = `[v2] ${el.dataset.eventCategory}`;
+    if (eventCategory.includes("Scan")) {
+      // Append user status to eventLabel for scan form events.
+      eventLabel = `${eventLabel} [Signed in user: ${document.body.dataset.signedInUser}]`;
+    }
+    return ga("send", "event", eventCategory, eventAction, eventLabel);
+  }
+};
+
+// Elements for which we send Google Analytics "View" pings...
+let eventTriggers = [
+  "#scan-user-email",
+  "#dl-fx-bar",
+  "#dl-fx-banner",
+  "#fxa-new-user-bar",
+  "#show-additional-breaches",
+  ".see-full-report-button",
+  ".open-oauth",
+  ".sign-up-button", // legacy sign up button, can be removed post-fxa
+];
+
+eventTriggers = document.querySelectorAll(eventTriggers);
+
+// Reset data-event-category and data-fxa-entrypoint if necessary when
+// the entrypoint is nested inside Sign Up Banner.
+// TODO: send top of funnel ping to FxA for each data-fxa-entrypoint element.
+document.querySelectorAll("#scan-user-email, .open-oauth").forEach(el => {
+  setMetricsIds(el);
+});
+
+const pageLocation = getLocation();
+
+// Send "View" pings and add event listeners.
+eventTriggers.forEach(el => {
+  sendPing(el, "View", pageLocation);
+  if (["BUTTON", "A"].includes(el.tagName)) {
```

Comment:
Is all-caps `tagName` cross-browser compatible? If not, might want to `el.tagName.toUpperCase()` first?